scripts/cert_generation.sh

```diff
@@ -10,34 +10,58 @@ REGENERATE_NEEDED=0
 
 function info() {
   local msg=$1
-  echo "[INFO] ${msg}" >> ${WORKING_DIR}/action.log
+  local type=${2:-}
+  local reason=${3:-}
+  local out="\"msg\":\"$msg\""
+  if [ -n "$type" ] ; then
+    out="$out, \"type\":\"$type\""
+  fi
+  if [ -n "$reason" ] ; then
+    out="$out, \"reason\":\"$reason\""
+  fi
+  if [ -f ${WORKING_DIR}/action.log ] && [ "$(stat -c%s ${WORKING_DIR}/action.log)" != "0" ] ; then
+    echo "," >> ${WORKING_DIR}/action.log
+  fi
+  echo "{$out}" >> ${WORKING_DIR}/action.log
 }
 
 function init_cert_files() {
 
   if [ ! -f ${WORKING_DIR}/ca.db ]; then
-    info "Initializing missing file: ${WORKING_DIR}/ca.db"
+    info "${WORKING_DIR}/ca.db" "File intialization" "Missing"
     touch ${WORKING_DIR}/ca.db
   fi
 
   if [ ! -f ${WORKING_DIR}/ca.serial.txt ]; then
-    info "Initializing missing file: ${WORKING_DIR}/ca.serial.txt"
+    info "${WORKING_DIR}/ca.serial.txt" "File intialization" "Missing"
     echo 00 > ${WORKING_DIR}/ca.serial.txt
   fi
 }
 
 function generate_signing_ca() {
   local regen=false
-  if [ ! -f ${WORKING_DIR}/ca.crt ] || [ "$(file -b ${WORKING_DIR}/ca.crt)" != "PEM certificate" ] ; then
-    info "Missing file or invalid type: ${WORKING_DIR}/ca.crt"
+  local fileExists=$(test -f ${WORKING_DIR}/ca.crt;echo $?)
+  if [ "$fileExists" == "1" ] ; then
```

Comment:
given we never use `fileExists` ever again after this, can we just keep it in the `if` block? needing to do an `echo $?` and then check that value manually seems like an anti-pattern here

Reply:
we use this in line 49.  I purposely added this distinction in order to get the singular reason for regen; missing versus filetype.  It otherwise logged both which did not pinpoint why